lms/djangoapps/instructor_task/tests/test_tasks_helper.py

```diff
@@ -1724,21 +1739,23 @@ def test_certificate_generation_whitelisted_not_generated(self):
             GeneratedCertificateFactory.create(
                 user=student,
                 course_id=self.course.id,
-                status=CertificateStatuses.downloadable,
+                status=status,
                 mode='honor'
             )
 
         # white-list 5 students
         for student in students:
-            CertificateWhitelistFactory.create(user=student, course_id=self.course.id, whitelist=True)
+            CertificateWhitelistFactory.create(
+                user=student, course_id=self.course.id, whitelist=True
+            )
 
         task_input = {'student_set': 'whitelisted_not_generated'}
 
         expected_results = {
             'action_name': 'certificates generated',
-            'total': 3,
-            'attempted': 3,
-            'succeeded': 3,
+            'total': expected_certs,
+            'attempted': expected_certs,
+            'succeeded': expected_certs,
             'failed': 0,
```

Comment:
@saleem-latif , do you mean there should be scenarios for these in general, or just for these whitelist tests? Because scenarios with skipped and failed do exist in this suite: https://github.com/edx/edx-platform/blob/master/lms/djangoapps/instructor_task/tests/test_tasks_helper.py#L1626-L1657